### PR TITLE
Add one.ablaze.floorp

### PR DESCRIPTION
**Pull request by nexryai**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

### CODE REVIEW

File: one.ablaze.floorp.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+command: floorp
+modules:                
+  - name: firefox
+    buildsystem: simple
+    build-commands:
+      - install -D root.tar.xz /app/root.tar.xz
+      - cd /app/ && tar Jxfv root.tar.xz && rm root.tar.xz
+    sources:
+      - type: file
+        path: root.tar.xz
```
[the remark this answers is not in the document]
Comment (nexryai) [reply one.ablaze.floorp.yml]: ok!
I fixed it

File: one.ablaze.floorp.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+    <id>one.ablaze.floorp</id>
+    <metadata_license>MPL-2.0</metadata_license>
```
Comment (hfiguiere) [inline one.ablaze.floorp.appdata.xml:4]: For the metadata license (THIS file) `CC0-1.0` is preferred.

File: one.ablaze.floorp.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        sha256: ee9a2c31385c56ea43cd699d1ed9569147cdd56695e567b7a374fa4f2639b1c4
+
+finish-args:
+  - --share=ipc
+  - --socket=pcsc  
+  - --socket=x11
+  - --socket=wayland
+  - --share=network
+  - --socket=pulseaudio
+  - --filesystem=home
```
Comment (hfiguiere) [inline one.ablaze.floorp.yml:32]: it's a web browser, it should have a more restricted access to the filesystem.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
The appdata file is not installed in your manifest.

also the appdata.xml should be submitted to be managed upstream to be included in a future release.

File: one.ablaze.floorp.appdata.xml (hunk 2 above)
Comment (hfiguiere) [reply one.ablaze.floorp.appdata.xml:4]: and it is missing the `project_license`

Comment (nexryai) [reply one.ablaze.floorp.appdata.xml]: ok
I fixed it

File: one.ablaze.floorp.yml (hunk 3 above)
Comment (hfiguiere) [reply one.ablaze.floorp.yml:16]: still

File: one.ablaze.floorp.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    build-commands:
+      - cd /app/ && tar Jxvf ${FLATPAK_BUILDER_BUILDDIR}/floorp.tar.xz
+      
+    sources:
+      - type: file
+        url: 'https://sda1.net/storage/floorp/tarball/8.7.2/floorp.tar.xz'
+        sha256: ee9a2c31385c56ea43cd699d1ed9569147cdd56695e567b7a374fa4f2639b1c4
+        
+      - type: file
+        path: org.kde.kst.appdata.xml
```
Comment (hfiguiere) [inline one.ablaze.floorp.yml:37]: that doesn't install it FWIW

Comment (hfiguiere) [reply one.ablaze.floorp.yml:37]: also it's the wrong name, obviously.

Comment (nexryai) [reply one.ablaze.floorp.yml]: Oops.
I'll fix it.

File: one.ablaze.floorp.yml (hunk 3 above)
Comment (nexryai) [reply one.ablaze.floorp.yml]: Changed to allow access only to the configuration and download directories.

File: one.ablaze.floorp.yml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    sources:
+      - type: file
+        url: 'https://sda1.net/storage/floorp/tarball/8.7.2/floorp.tar.xz'
+        sha256: ee9a2c31385c56ea43cd699d1ed9569147cdd56695e567b7a374fa4f2639b1c4
+
+finish-args:
+  - --share=ipc
+  - --socket=pcsc  
+  - --socket=x11
+  - --socket=wayland
```
[the remark this answers is not in the document]
Comment (nexryai) [reply one.ablaze.floorp.yml]: Fixed in https://github.com/flathub/flathub/pull/3103/commits/e25ee120c3a88daa28e0d48413d351516c7f30c9

File: one.ablaze.floorp.yml (hunk 4 above)
Comment (nexryai) [reply one.ablaze.floorp.yml]: Fixed in https://github.com/flathub/flathub/pull/3103/commits/e25ee120c3a88daa28e0d48413d351516c7f30c9

File: one.ablaze.floorp.yml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        sha256: ee9a2c31385c56ea43cd699d1ed9569147cdd56695e567b7a374fa4f2639b1c4
+        
+  - name: appdata
+    buildsystem: simple
+    build-commands:
+      - cp ${FLATPAK_BUILDER_BUILDDIR}/one.ablaze.floorp.appdata.xml /app/share/metainfo/
+      
+    sources:
+      - type: file
+        path: one.ablaze.floorp.appdata.xml
```
Comment (hfiguiere) [inline one.ablaze.floorp.yml:43]:
```suggestion
  - name: floorp
    buildsystem: simple
    build-commands:
      - install -Dm644 one.ablaze.floorp.appdata.xml -t /app/share/metainfo/      
      - cd /app/ && tar Jxvf ${FLATPAK_BUILDER_BUILDDIR}/floorp.tar.xz
    sources:
      - type: file
        url: 'https://sda1.net/storage/floorp/tarball/8.7.2/floorp.tar.xz'
        sha256: ee9a2c31385c56ea43cd699d1ed9569147cdd56695e567b7a374fa4f2639b1c4
      - type: file
        path: one.ablaze.floorp.appdata.xml
```

File: one.ablaze.floorp.yml [hunk 7]
```diff
@@ -0,0 +1,45 @@
+app-id: one.ablaze.floorp
```
Comment (hfiguiere) [inline one.ablaze.floorp.yml:1]: the ustrem tarball use `org.ablaze.floorp` to install icons and the desktop file. And that `.org` domain seem to not exist. so renaming need to be done. (and possibly changes to rename them submitted upstream along with the appdata file)

File: one.ablaze.floorp.yml [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+command: floorp
+
+finish-args:
+  - --share=ipc
+  - --socket=pcsc  
+  - --socket=x11
+  - --share=network
+  - --socket=pulseaudio
+  - --filesystem=~/.mozilla
```
Comment (hfiguiere) [inline one.ablaze.floorp.yml:15]: should this be `--persist` instead?

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
bot, build one.ablaze.floorp

**Comment by barthalion** [comment]
Build failed because metainfo directory is missing:

```
cp: cannot create regular file '/app/share/metainfo/': Not a directory
```

File: one.ablaze.floorp.yml (hunk 6 above)
Comment (nexryai) [reply one.ablaze.floorp.yml]: Thanks!

File: one.ablaze.floorp.yml (hunk 8 above)
Comment (nexryai) [reply one.ablaze.floorp.yml]: fixed in https://github.com/flathub/flathub/pull/3103/commits/91776956fc4b0520a58ae9a7e854960060d49b29

File: one.ablaze.floorp.yml (hunk 7 above)
Comment (nexryai) [reply one.ablaze.floorp.yml:1]: OK
Updated upstream tarball and updated checksums.

**Comment by hfiguiere** [comment]
bot, build one.ablaze.floorp

**Comment by hfiguiere** [comment]
the .desktop in the tarball is wrong in two ways.

1. the icon name is wrong, **and that's the build failure**
2. `Exec` is not a path, just the name of the command

**Comment by barthalion** [comment]
bot, build one.ablaze.floorp

**Pull request closed without merging**